tools/unit_tests.py

```diff
@@ -10,25 +36,12 @@
 # tests by the special string. permW0N0 means allow-write but not allow-net.
 # See js/test_util.ts for more details.
 def unit_tests(deno_exe):
-    run([deno_exe, "--reload", "js/unit_tests.ts", "permW0N0E0"])
-    run([
-        deno_exe, "--reload", "js/unit_tests.ts", "permW1N0E0", "--allow-write"
-    ])
-    run([
-        deno_exe, "--reload", "js/unit_tests.ts", "permW0N1E0", "--allow-net"
-    ])
-    run([
-        deno_exe, "--reload", "js/unit_tests.ts", "permW0N0E1", "--allow-env"
-    ])
-    run([
-        deno_exe,
-        "--reload",
-        "js/unit_tests.ts",
-        "permW1N1E1",
-        "--allow-write",
-        "--allow-net",
-        "--allow-env",
-    ])
+    run_unit_test(deno_exe, "permW0N0E0")
+    run_unit_test(deno_exe, "permW1N0E0", ["--allow-write"])
+    run_unit_test(deno_exe, "permW0N1E0", ["--allow-net"])
+    run_unit_test(deno_exe, "permW0N0E1", ["--allow-env"])
+    # TODO We might accidentally miss some. We should be smarter about which we
```

Comment:
Yeah i worry about that too.

We could also do some typescript trick with the `DenoPermissions` type in the test runner so it only allows the 4 permutations listed here. Then it'd be obvious when a new test uses a flag combination that doesn't get run.

```ts
type Exact<A extends object> = A & { __kind: keyof A };
type DenoPermissions =
  Exact<{} | { write: true } | { net: true } | { env: true }>;
```